### ISSUE TITLE
[4.0.x] Segfault session start

### ISSUE THREAD

**Issue by jturbide** [phalcon/cphalcon#13718]
### Expected and Actual Behavior

> **Describe what you are trying to achieve and what goes wrong.**

[4.0.x] Trying to start the session, getting a segfault

> Provide output if related. Provide coredump if any. Use https://docs.phalconphp.com/en/latest/generating-backtrace as reference

```php
BACKTRACE TBD
```

```php
--- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=0x9} ---
+++ killed by SIGSEGV +++
Segmentation fault
```

> Provide minimal script to reproduce the issue

```php
$session = new \Phalcon\Session\Manager();
$handler = new \Phalcon\Session\Adapter\Noop();
$session->setHandler($handler);
$session->start();
```

### Details

```
# php --ri phalcon

phalcon


Web framework delivered as a C-extension for PHP
phalcon => enabled
Author => Phalcon Team and contributors
Version => 4.0.0-alpha1
Build Date => Dec 28 2018 12:00:04
Powered by Zephir => Version 0.11.9-$Id$

Directive => Local Value => Master Value
phalcon.db.escape_identifiers => On => On
phalcon.db.force_casting => Off => Off
phalcon.orm.events => On => On
phalcon.orm.virtual_foreign_keys => On => On
phalcon.orm.column_renaming => On => On
phalcon.orm.not_null_validations => On => On
phalcon.orm.exception_on_failed_save => Off => Off
phalcon.orm.enable_literals => On => On
phalcon.orm.late_state_binding => Off => Off
phalcon.orm.enable_implicit_joins => On => On
phalcon.orm.cast_on_hydrate => Off => Off
phalcon.orm.ignore_unknown_columns => Off => Off
phalcon.orm.update_snapshot_on_save => On => On
phalcon.orm.disable_assign_setters => Off => Off
phalcon.orm.case_insensitive_column_map => Off => Off

```

```
php -v
PHP 7.2.13 (cli) (built: Dec  8 2018 12:11:34) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with the ionCube PHP Loader (enabled) + Intrusion Protection from ioncube24.com (unconfigured) v10.3.0, Copyright (c) 2002-2018, by ionCube Ltd.
    with Zend OPcache v7.2.13, Copyright (c) 1999-2018, by Zend Technologies
    with Xdebug v2.6.1, Copyright (c) 2002-2018, by Derick Rethans
```

```
# uname -a
Linux php.jturbide.com 3.10.0-957.1.3.el7.x86_64 #1 SMP Thu Nov 29 14:49:43 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

```
# cat /etc/*-release
CentOS Linux release 7.6.1810 (Core)
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

CentOS Linux release 7.6.1810 (Core)
CentOS Linux release 7.6.1810 (Core)
```

```
# zephir --version
Zephir 0.11.9 by Andres Gutierrez and Serghei Iakovlev (source)
```

```
# php -r 'foreach (get_loaded_extensions() as $extension) echo "$extension: " . phpversion($extension) . "\n";'
Core: 7.2.13
date: 7.2.13
libxml: 7.2.13
openssl: 7.2.13
pcre: 7.2.13
zlib: 7.2.13
filter: 7.2.13
hash: 1.0
pcntl: 7.2.13
readline: 7.2.13
Reflection: 7.2.13
SPL: 7.2.13
session: 7.2.13
standard: 7.2.13
bz2: 7.2.13
calendar: 7.2.13
ctype: 7.2.13
curl: 7.2.13
dom: 20031129
mbstring: 7.2.13
fileinfo: 1.0.5
ftp: 7.2.13
gd: 7.2.13
gettext: 7.2.13
gmp: 7.2.13
iconv: 7.2.13
intl: 1.1.0
json: 1.6.0
exif: 7.2.13
mysqlnd: mysqlnd 5.0.12-dev - 20150407 - $Id: 38fea24f2847fa7519001be390c98ae0acafe387 $
PDO: 7.2.13
Phar: 2.0.2
posix: 7.2.13
shmop: 7.2.13
SimpleXML: 7.2.13
sockets: 7.2.13
sodium: 7.2.13
sqlite3: 7.2.13
sysvmsg: 7.2.13
sysvsem: 7.2.13
sysvshm: 7.2.13
tokenizer: 7.2.13
xml: 7.2.13
xmlwriter: 7.2.13
xsl: 7.2.13
mcrypt: 1.0.1
mysqli: 7.2.13
pdo_mysql: 7.2.13
pdo_sqlite: 7.2.13
wddx: 7.2.13
xmlreader: 7.2.13
apcu: 5.1.16
psr: 0.6.1
zip: 1.15.4
phalcon: 4.0.0-alpha1
Zephir Parser: 1.2.0
ionCube Loader:
Zend OPcache: 7.2.13
xdebug: 2.6.1

```

**Comment by sergeyklay** [comment]
I'll try to sort out ASAP

**Comment by sergeyklay** [comment]
Fixed in the `4.0.x` branch. Feel free to open a new issue if the problem appears again. Thank you for contributing.
